presidio-cli/presidio_cli/cli.py

```diff
@@ -0,0 +1,282 @@
+
+
+def find_files_recursively(
+    items: List[str],
+    conf: PresidioCLIConfig
+) -> Generator[str, None, None]:
+    """
+    Generate all file names inside the directories.
+
+    :param items: List of directories containing files to be analyzed.
+    :param conf: PresidioCLIConfig object
+    """
+    for item in items:
+        if os.path.isdir(item):
+            for root, dirnames, filenames in os.walk(item):
+                for f in filenames:
+                    filepath = os.path.join(root, f)
+                    if conf.is_text_file(filepath):
+                        yield filepath
+        else:
+            if conf.is_text_file(item):
+                yield item
+
+
+def run() -> None:
+    """
+    Entrypoint of Presidio CLI.
+    """
+    parser = argparse.ArgumentParser(prog=SHELL_NAME, description=APP_DESCRIPTION)
+
+    parser.add_argument("-v", "--version", action="version", version=f"v{APP_VERSION}")
+
+    files_group = parser.add_mutually_exclusive_group(required=True)
+    files_group.add_argument(
+        "files",
+        metavar="FILE_OR_DIR",
+        nargs="*",
+        default=(),
+        help="files to check",
+    )
+    files_group.add_argument(
+        "-", action="store_true", dest="stdin", help="read from standard input"
+    )
+
+    config_group = parser.add_mutually_exclusive_group()
+    config_group.add_argument(
+        "-c",
+        "--config-file",
+        dest="config_file",
+        action="store",
+        help="path to a custom configuration",
+    )
+
+    config_group.add_argument(
+        "-d",
+        "--config-data",
+        dest="config_data",
+        action="store",
+        help="custom configuration (as YAML source)",
+    )
+
+    parser.add_argument(
+        "-f",
+        "--format",
+        choices=("standard", "github", "auto", "colored", "parsable"),
+        default="auto",
+        help="format for parsing output",
+    )
+
+    parser.add_argument(
+        "--no-warnings",
+        action="store_true",
+        help="output only error level problems",
+    )
+
+    args = parser.parse_args()
+
+    try:
+        if args.config_data is not None:
+            if args.config_data != "" and ":" not in args.config_data:
+                args.config_data = "extends: " + args.config_data
+            conf = PresidioCLIConfig(content=args.config_data)
+        elif args.config_file is not None:
+            conf = PresidioCLIConfig(file=args.config_file)
+        elif os.path.isfile(".presidiocli"):
+            conf = PresidioCLIConfig(file=".presidiocli")
+        else:
+            conf = PresidioCLIConfig(content="extends: default")
+    except PresidioCLIConfigError as e:
+        print(e, file=sys.stderr)
+        sys.exit(1)
+
+    if conf.locale is not None:
+        locale.setlocale(locale.LC_ALL, conf.locale)
+
+    prob_num = 0
+    for file in find_files_recursively(args.files, conf):
+        filepath = file[2:] if file.startswith("./") else file
```

Comment:
Would `file.startswith("./")` work on Windows too?

Reply:
I've added a condition that handles Windows paths (starting with `.\`). However, even without that condition, this worked fine (`filepath` starting with `.\`).